cmd/main.go

```diff
@@ -144,6 +144,21 @@ func runBuild(ctx context.Context, args []string) error {
 		log.Fatalf("[FATAL] %v", err)
 	}
 
+	if os.Getenv("XCADDY_SETCAP") == "1" {
+		var cmd *exec.Cmd
+		if os.Getenv("XCADDY_SETCAP_NO_SUDO") == "1" {
+			cmd = exec.Command("setcap", "cap_net_bind_service=+ep", output)
+		} else {
+			cmd = exec.Command("sudo", "setcap", "cap_net_bind_service=+ep", output)
+		}
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		log.Printf("[INFO] Setting capabilities (requires admin privileges): %v", cmd.Args)
+		if err = cmd.Run(); err != nil {
+			return err
```

Comment:
The `sudo` failure at this step will leave the built binary behind without the capability (which is ok) but the execution ends with:
```
023/01/24 19:41:59 [INFO] Setting capabilities (requires admin privileges): [sudo setcap cap_net_bind_service=+ep custom-caddy]
[sudo] password for mohammed:
Sorry, try again.
[sudo] password for mohammed:
Sorry, try again.
[sudo] password for mohammed:
sudo: 3 incorrect password attempts
2023/01/24 19:43:20 [ERROR] exit status 1
```
Can the error be more descriptive?

Reply:
`sudo` doesn't exist in the container:

```
$ docker run --rm -it golang:1.19-alpine /bin/sh
/go # sudo setcap
/bin/sh: sudo: not found
```

Good point regarding the error message, I'll fix that up.